### ISSUE TITLE
Networkd doesn't restart anymore on unit changes in 20.03

### ISSUE THREAD

**Issue by arianvp** [NixOS/nixpkgs#85800]
**Describe the bug**
When you change a networkd option, networkd doesn't pick them up anymore

**To Reproduce**
Change a `networkd` option and see it not getting applied

Bisected to https://github.com/NixOS/nixpkgs/commit/1d61efb7f177

Bug was introduced here: https://github.com/NixOS/nixpkgs/pull/63103/files?file-filters%5B%5D=.nix#diff-7d08c03fc593d925ed022e395579c866R941

It seems to replace `x.source` with `attrName` in `restartTriggers` **however this is a semantic change. restartTriggers should point to derivations, not strings. This breaks  things**

**Screenshots**

**Additional context**

**Notify maintainers**

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by arianvp** [comment]
That PR seems to not only have broken `networkd` but many other services as well. We should have a closer look

**Comment by arianvp** [comment]
The only other affected service seems to be `datadog` https://github.com/NixOS/nixpkgs/pull/63103/files?file-filters%5B%5D=.nix#diff-05b73b65b3562a8474e78df934f044c2R228

I'll prepare a PR and a backport PR

**Comment by flokli** [comment]
The fix has been merged to master and 20.09.
